### PR TITLE
Suppress Opflow deprecation warnings in test_qpy.py

### PR DESCRIPTION
**Pull request by t-imamichi**
### Summary

Removes opflow in test_qpy.py due to #9176
https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=45729&view=logs&j=e4ea27c2-3a16-5b32-7be8-0bf30fe9e828&t=b4382375-beb9-5aee-c896-08d1279c743b

### CODE REVIEW

**Review by jakelishman — COMMENTED** [review]
These deprecation warnings are fine, though would it be a good idea for us to start including the tests of what will be the new form of the functionality as well?  I don't know if we have coverage of that, yet.

File: test/qpy_compat/test_qpy.py [hunk 1]
```diff
@@ -130,6 +130,9 @@ def generate_random_circuits():
 
 def generate_string_parameters():
     """Generate a circuit from pauli tensor opflow."""
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", DeprecationWarning)
+        from qiskit.opflow import X, Y, Z
     op_circuit = (X ^ Y ^ Z).to_circuit_op().to_circuit()
     op_circuit.name = "X^Y^Z"
     return op_circuit
```
Comment (ElePT) [inline test/qpy_compat/test_qpy.py:138]: I think this method just used opflow as a convenient way to generate a quantum circuit, so we actually don't use opflow for anything QPY-related, why don't we replace it with the new version already?
```suggestion
def generate_string_parameters():
    """Generate a circuit for the XYZ pauli string."""
    op_circuit = QuantumCircuit(3, name="X^Y^Z")
    op_circuit.z(0)
    op_circuit.y(1)
    op_circuit.x(2)
    return op_circuit
```

File: test/qpy_compat/test_qpy.py [hunk 2]
```diff
@@ -330,6 +333,10 @@ def generate_evolution_gate():
     from qiskit.circuit.library import PauliEvolutionGate
     from qiskit.synthesis import SuzukiTrotter
 
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", DeprecationWarning)
+        from qiskit.opflow import I, Z
+
     synthesis = SuzukiTrotter()
     evo = PauliEvolutionGate([(Z ^ I) + (I ^ Z)] * 5, time=2.0, synthesis=synthesis)
```
Comment (ElePT) [inline test/qpy_compat/test_qpy.py:341]: Here, we are again not testing anything QPY-related using opflow, the `PauliEvolutionGate` tests already have both cases, so maybe we could also just get rid of the dependency. In that case, the alternative would be:

```suggestion
    synthesis = SuzukiTrotter()
    op = SparsePauliOp.from_list([('ZI', 1), ('IZ', 1)])
    evo = PauliEvolutionGate([op] * 5, time=2.0, synthesis=synthesis)
```

File: test/qpy_compat/test_qpy.py [hunk 3]
```diff
@@ -330,6 +333,10 @@ def generate_evolution_gate():
     from qiskit.circuit.library import PauliEvolutionGate
     from qiskit.synthesis import SuzukiTrotter
```
Comment (ElePT) [inline test/qpy_compat/test_qpy.py:334]: Import for suggestion below:
```suggestion
    from qiskit.circuit.library import PauliEvolutionGate
    from qiskit.synthesis import SuzukiTrotter
    from qiskit.quantum_info import SparsePauliOp
```

**Review by ElePT — COMMENTED** [review]
Sorry @t-imamichi that you have to clean up again after the opflow deprecation PR. As I explain in the specific comments, I think that we could already get rid of the opflow dependency in this test in particular.

**Comment by t-imamichi** [comment]
Thank you for your suggestions. I applied them.

**Review by ElePT — APPROVED** [review]
LGTM!

File: test/qpy_compat/test_qpy.py [hunk 4]
```diff
@@ -129,9 +128,11 @@ def generate_random_circuits():
...
 def generate_string_parameters():
-    """Generate a circuit from pauli tensor opflow."""
-    op_circuit = (X ^ Y ^ Z).to_circuit_op().to_circuit()
-    op_circuit.name = "X^Y^Z"
+    """Generate a circuit for the XYZ pauli string."""
+    op_circuit = QuantumCircuit(3, name="X^Y^Z")
+    op_circuit.z(0)
+    op_circuit.y(1)
+    op_circuit.x(2)
     return op_circuit
```
Comment (jakelishman) [inline test/qpy_compat/test_qpy.py:136]: I think this change is actually breaking the intended test; in the original, there's a `PauliGate` in the circuit, which has a `.params` field that contains a string.  The new form doesn't have that; it's just got a string in the `op_circuit` name, which is the standard.

We could probably hew to the intent of the test with something like
```python
from qiskit.circuit.library import PauliGate

op_circuit = QuantumCircuit(3)
op_circuit.append(PauliGate("XYZ"), op_circuit.qubits, [])
```

Comment (t-imamichi) [reply test/qpy_compat/test_qpy.py]: Thanks. I use PauliGate 97f5a84.

**Review by jakelishman — APPROVED** [review]
Thanks for the changes, this should make the tests a little less noisy now.

**Pull request merged**